cardano-testnet/src/Testnet/Util/Cli.hs

```diff
@@ -0,0 +1,186 @@
+module Testnet.Util.Cli
+  ( cliAddressKeyGen
+  , cliNodeKeyGen
+  , cliNodeKeyGenVrf
+  , cliNodeKeyGenKes
+  , cliStakeAddressKeyGen
+  , Comment (..)
+  , KeyNames (..)
+
+  , fakeItH
+  , fakeIt
+  , TmpDir
+  , TestnetMagic
+  , File (..)
+  , Address
+  , Operator -- == Cold Keys
+  , Kes
+  , StakeAddress
+  , Vrf
+
+  , VKey
+  , SKey
+
+  , OperatorCounter
+  , ByronKey
+  , ByronAddress
+
+  , ByronDelegationKey
+  , ByronDelegationCert
+
+  , getVKeyPath
+  , getSKeyPath
+
+  , makeFilePath
+  , cliKeyGen
+
+  , cliSigningKeyAddress
+  ) where
+
+import           Prelude
+
+import           Data.String
+import           Control.Monad
+import           System.FilePath.Posix
+
+import qualified Hedgehog.Extras.Test.Base as H
+import qualified Hedgehog.Extras.Test.File as H (writeFile)
+import           Hedgehog
+import           Hedgehog.Gen
+
+import           Testnet.Util.Process
+
+-- TODO: check no whitespace , slash etc allowed
+-- TODO: use Comment instead of Filepath
+newtype Comment = Comment { unComment :: String}
+  deriving (Show, Eq, Ord )
+
+instance IsString Comment where fromString a = Comment a
+
+instance Semigroup Comment where
+  (Comment a) <> (Comment b) = Comment $ a <> "-" <> b
+
+-- Could also use IO (timestamp).
+makeFilePath :: MonadGen m => FilePath -> Comment -> FilePath -> m FilePath
+makeFilePath path comment suffix = do
+  almostUnique <- Comment <$> replicateM 4 hexit
+  return $ path </> unComment ( comment <> almostUnique) <.> suffix
+
+data KeyNames = KeyNames
+  { verificationKeyFile :: FilePath
+  , signingKeyFile :: FilePath
+  }
+
+type KeyGen a = H.Integration (File (a VKey), File (a SKey))
```

Comment:
The `KeyGen` type alias seems less helpful.

Reply:
I do not know.. I looked up the code `KeyGen` is used five times.